### ISSUE TITLE
help page broken with "-a" on a restricted shell

### ISSUE THREAD

**Issue by mc-butler** [MidnightCommander/mc#4301]
> [!IMPORTANT]
> This issue was migrated from Trac:
>
> | **Origin** | https://midnight-commander.org/ticket/4301 |
> | :------------- | :------------------------------------------------------------------------------------------------ |
> | **Reporter** | `alexander.stohr` (alexander.stohr@….de) |

<p>
use a shell that does not honor extended escape sequences (found this when using <tt>TeraTerm</tt> and a SSH connection to some ARM64 embedded board).<br />
mc version on the target system: 4.8.19<br />
</p>
<p>
as the border frames are "corrupted" i needed to use the command "mc -a" to launch the software and getting a useable screen.<br />
</p>
<p>
when calling help screen i found the big MC logo there to not displaying correctly. also the right border of the help text window is corrupted.<br />
</p>

> [!NOTE]
> Original attachments:
>
> * [mc-help-corruption.png](https://github.com/MidnightCommander/trac-archive/blob/master/attachments/ticket/4301/mc-help-corruption.png) ([raw](https://github.com/MidnightCommander/trac-archive/raw/master/attachments/ticket/4301/mc-help-corruption.png)) by `alexander.stohr` (alexander.stohr@….de) on <span title="2021-11-12T13:21:01+00:00">`Nov 12, 2021 at 13:21 UTC`</span>

**Comment by egmontkob** [comment]
This really looks like a locale mismatch to me.

The terminal seems to run with Windows-1252 (or some very similar) encoding, whereas mc sees a UTF-8 locale. These line drawing characters are all 3 bytes in UTF-8. The first byte in all of them is 226 which appears as `â` (as seen all over the screenshot). The third byte of the horizontal line U+2500 is 128, which is the `€` symbol (seen e.g. at the 6th broken position in the screenshot, i.e. the last byte of the second desired character).

It's not a limitation of the terminal (let alone the shell, it has nothing to do with it), it's a misconfiguration of the system, a broken setup. The proper solution is not `mc -a` but to fix the locale.

---

That being said, it would indeed be nice if `-a` also converted that logo to stickchars.

This should be quite easy (famous last words...) on top of pending PR #4813. Stickchars shouldn't be handled on the skin level, overwriting whatever chars the skin defines; it should be handled on the tty level, slang and ncurses emitting these chars instead of the ACS stuff.

**Comment by egmontkob** [comment]
It would be quite ugly though ... well, internally "stickchars" is called "ugly lines" for a reason :)

```
 +---+     +           +                         
 | | | .   |     .     |                         
 | | | + +-| --+ + +-- |-+ +                     
 | | | | | | | | | | | | | |                     
 -   - - +-- - - - +-| - - ++                    
     ------Commander-+ 4.8.33
```

(no code yet, just search-and-replaced)

**Comment by egmontkob** [comment]
The ugly part will be that `-a` stickchars (ugly lines) and `-s` slow terminals (spaces) can't be handled analogously, at the same place.

For the very reason that we want stickchars to apply to the logo, but we don't want it to vanish in case of slow terminals.

So while handling stickchars moves to the tty layer, handling slow terminals remains in the skin (pretending the skin defined spaces for all the frames, not affecting the mc logo).

[mc-4301-ugly-logo.patch](https://github.com/user-attachments/files/23145287/mc-4301-ugly-logo.patch) -- goes on top of PR 4813.

**Comment by egmontkob** [comment]
> The ugly part will be

No, it won't. I realized it's the wrong approach to handle these in the tty layer. That's because with stickchars I assume we'd like to have stickchars even if the skin defines completely different characters. So we indeed have to override the skin (as we do now), not the presentation of the line drawing characters.

The fix becomes much simpler: just make the help viewer convert its own ACS chars to these stickchars.

Continuing in PR #4813...
